vault/identity_store_util.go

```diff
@@ -0,0 +1,1244 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	memdb "github.com/hashicorp/go-memdb"
+	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/consts"
+	"github.com/hashicorp/vault/helper/locksutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+func (c *Core) loadEntities() error {
+	if c.identityStore == nil {
+		return fmt.Errorf("identity store is not setup")
+	}
+
+	return c.identityStore.loadEntities()
+}
+
+func (i *identityStore) loadEntity(entity *entityStorageEntry, previousEntity *entityStorageEntry) error {
```

Comment:
Can you add some comments here explaining `previousEntity`? I know what it does from the review but would be good for future reviewers/developers.